### ISSUE TITLE
Relation between Conformance + Landing Page

### ISSUE THREAD

**Issue by m-mohr** [opengeospatial/oapi_common#71]
For me it is not clear how the relation between the landing page `/` and the conformance page `/conformance` is. It seems like a client should connect to `/` and then request `/conformance`.

1. As the response of conformance is pretty simple, I'm wondering why it is not part of `/`.
2. At the moment it seems already that `/conformance` may have implications how I should read `/`. Would a "xml" conformance mean that the landing page is xml?
3. If I want to extent the landing page with whatever extension (there's already use cases in STAC and openEO), I would need to add that extension to `/conformance` just to figure out that I may need to revisit `/`, because it contains more than initially thought.

To avoid this, I really think **`/conformance` should be merged into `/`** so that on the initial request to the landing page I already know what the service conforms to. I guess a client is not expected to visit `/conformance` before `/`, right?

**Comment by joanma747** [comment]
Only OGC API clients will be interested in the conformance. 
The "follow the links" approach works even if you are not reading the conformance.
Not that the https://github.com/opengeospatial/oapi_common/blob/master/core/requirements/core/REQ_conformance-op.adoc states that "/conformance" is a mandatory path. If you read the same for the API description you will see that /api is not mandatory.   

A OGC API client will need to read both resources to know what to do exactly.

Resolution:

Add a description on why /conformance has a "fix" path but the others do not (based on the start of this discussion and the text provided by @m-mohr.

State clearly that only "/" and "/conformance" are the only resources that have a mandatory "path" and are URLs.

Also fix the first part of: https://github.com/opengeospatial/oapi_common/blob/master/core/clause_8_core.adoc and change : 
2.- "/api" - the API Definition document for this API
to
2.- the API Definition document for this API (commonly in "/api")

**Comment by joanma747** [comment]
WE still have consensus and pending of applying this to the document. 
The issue is assigned to: @cmheazel

**Comment by cmheazel** [comment]
Re-wrote section 8.3 providing an explanation of why the Conformance Class resource is designed the way it is. 
Did not change the /api description. API-Common conflates the use of path as part of an operation with path as an identifier. 
    /api as an identifier is independent of any implementation
    /api as an OpenAPI Path element describes how to access the /api resource on a specific API implementation.
By untangling identifiers and operations we can provide a clearer and more coherent set of standards.

**Comment by cmheazel** [comment]
Recommend closing since the unresolved issue is covered by issue #60.

**Comment by jerstlouis** [comment]
I would agree with @m-mohr on this one, but for Common to stay compatible with the published Features standard things have to remain as they are. Clients not interested in the conformance could have simply ignored the conformance property.
It is awkward that a client has to make so many round-trips to the server (/, /conformance, /collections) just to get e.g. the list of collections.

**Comment by m-mohr** [comment]
@jerstlouis The other question is: Is Common really staying fully compatible with Features? If not (thinking about #76 and #91, for example), this could also be changed. Because if there are breaking changes anyway, there's no point in the "compatibility" argument any longer.

@cmheazel I've read the new 8.3 (I hope it's the one in master):  

Some comments:

>  The first question a client will ask when accessing one of these APIs is "what are you?"The first question a client will ask when accessing one of these APIs is "what are you?"

So why is /conformance only the second thing to request if it's the "first question a client will ask"?

What I'm also missing from the description in 8.3: There's no clear indication on whether the things in /conformance apply to / or not? Do I need to handle / again after reading /conformance? Does it depends on what /conformance specifies?

I'm wondering: You have to specify a link in / with relation type "conformance" that points (usually) to "/conformance". Now that people discuss "non-fixed" paths #152 it seems, couldn't someone just point that link to "/" and merge the responses? I see it's not allowed in 8.3, but I'm not sure I follow the arguments there as you need to read and parse / anyway?!

**Comment by m-mohr** [comment]
via e-mail:

> Please look through them and identify any which you think should not be closed. We will discuss those next Monday. The rest will be closed after Monday’s meeting.

There are open questions above.

**Comment by cmheazel** [comment]
@m-mohr No, the new content is in the August 2020 branch. Master has not been updated.
`/conformance` is one of the few fixed paths. That allows you to go straight to the list of conformance classes. 
Once you have the conformance classes, and if the API implements the recommended fixed paths (perhaps a new conformance class?) you can go directly to any resource.

**Comment by m-mohr** [comment]
> I would agree with @m-mohr on this one, but for Common to stay compatible with the published Features standard things have to remain as they are. 

The compliance seems to be obsolete as the relation types have been changed in a breaking manor (e.g. conformance to ogc/conformance). Thus, I think the compatibility argument is not really valid any longer. But I'm not sure: Some parts of the docs still refer to the rel type `conformance` and some use `ogc/conformance`. #161

> `/conformance` is one of the few fixed paths. That allows you to go straight to the list of conformance classes.

Okay, I read the August branch now. It's still not clear from the docs whether the /conformance should be accessed first. Is it up to the client? Is there a best practice? From the word "landing page" for / I'd understand it's the first page to request, but then the landing page is influenced by /conformance and it seems that should be the first page to request so that you don't need two round trips to the landing page. I still think it should be part of the landing page, but at least it should be made clear how clients should behave.

**Comment by cmheazel** [comment]
@m-mohr I don't think it's appropriate for a standard to dictate how a client should use a Web API. There is no reason I cannot access `/collections/{collectionId}/items/{itemid}` without going through the landing page. That being said, instructions on how the Web API can be used are useful. This material should reside in the Users Guide (OGC 20-071).

**Comment by cmheazel** [comment]
@m-mohr An alternate pattern:
1)  I receive the URI to the OpenAPI document on a Web API (`/api`)
2) I process the OpenAPI document using Kaizen
3) I now have a set of Java classes which allow me to directly access any path (including the appropriate HTTP operations) on the API. I also have a full definition of all parameters.
4) I can now use the API without ever touching the landing page or any of the relationships.

**Comment by m-mohr** [comment]
As a developer, I'd expect that after reading a standard it's clear to me how to access the API properly. During implementation, the request order was not clear to me and left me confused. Leaving it all open to interpretation, may result in different behavior and decreased interoperability.

The alternate pattern seems overly complex and also bound to a specific language. I don't program Java at the moment. I'd just follow the links in the landing page, if it's JSON. But I don't even know whether I can expect JSON without calling /conformance first.

**Comment by joanma747** [comment]
I believe the flow for a generic client to use a OGC API - Common compatible looks like this.
* Visit the /conformance assuming that json will be returned
* If the response returns a link supporting OpenAPI
  - Visit the Landing page to find the OpenAPI link with the link relation appropriate
  - Visit the OpenAPI document and use it.
* If the response does provide a API description that the client does not know
  - Visit the Landing page to find links that can be used directly and start discovering resources.

We could try to add some flows like this one in Part 1.

**Comment by ghobona** [comment]
2020-08-24 SWG telecon resolution.

Capture in the Users' Guide a discussion of how to use Web APIs and make sure that all versions or parts of OGC API - Common draw the user's attention to that information.

Additional resolution. Recommendation to the DocTeam to define:

* a User Guide as a comprehensive collection of information for the developer
* a Getting Started Guide as the general information that a user may need to know to get started using the standard
